lib/daru/date_time/index.rb

```diff
@@ -268,6 +268,16 @@ def initialize *args
       @periods   = data.size
     end
 
+    # dup method for DateTimeIndex
+    
+    def dup
+      data = []
+      @data.each do |i|
+        data << i[0]
+      end
+      Daru::DateTimeIndex.new(data, :freq => @offset)
```

Comment:
You can just write this as `Daru::DateTimeIndex.new(@data.transpose[0], :freq => @offset)`